Lib/test/test_capi/test_long.py

```diff
@@ -34,6 +34,28 @@ def test_compact_known(self):
         self.assertEqual(_testcapi.call_long_compact_api(sys.maxsize),
                          (False, -1))
 
+    def test_long_asint(self):
+        PyLong_AsInt = _testcapi.PyLong_AsInt
+        INT_MIN = _testcapi.INT_MIN
+        INT_MAX = _testcapi.INT_MAX
+
+        # round trip (object -> int -> object)
+        for value in (INT_MIN, INT_MAX, -1, 0, 1, 123):
+            with self.subTest(value=value):
+                self.assertEqual(PyLong_AsInt(value), value)
+
+        # bound checking
+        with self.assertRaises(OverflowError):
+            PyLong_AsInt(INT_MIN - 1)
+        with self.assertRaises(OverflowError):
+            PyLong_AsInt(INT_MAX + 1)
+
+        # invalid type
```

Comment:
Please add also tests for objects with `__index__`

Reply:
Ok, I added a test